src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,124 @@
+    private static final Integer VALID_AGE_VALUE = 18;
+    private RegistrationService registrationService;
+    private StorageDao storageDao;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_nullLoginValue_throwsException() {
+        User user = createUser(null, VALID_PASSWORD_VALUE, VALID_AGE_VALUE);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_shortLoginValue_throwsException() {
+        User user = createUser("abc", VALID_PASSWORD_VALUE, VALID_AGE_VALUE);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_edgeLoginValue_throwsException() {
+        User user = createUser("abc12", VALID_PASSWORD_VALUE, VALID_AGE_VALUE);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_validLoginValue_success() {
+        User user = createUser(VALID_LOGIN_VALUE, VALID_PASSWORD_VALUE, VALID_AGE_VALUE);
+        registrationService.register(user);
+        assertTrue(storageDao.get(user.getLogin()) != null);
+    }
+
+    @Test
+    void register_nullPasswordValue_throwsException() {
+        User user = createUser(VALID_LOGIN_VALUE, null, VALID_AGE_VALUE);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_shortPasswordValue_throwsException() {
+        User user = createUser(VALID_LOGIN_VALUE, "abc", VALID_AGE_VALUE);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_edgePasswordValue_throwsExeption() {
+        User user = createUser(VALID_LOGIN_VALUE, "abc12", VALID_AGE_VALUE);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_validPasswordValue_success() {
+        User user = createUser(VALID_LOGIN_VALUE, VALID_PASSWORD_VALUE, VALID_AGE_VALUE);
+        registrationService.register(user);
+        assertTrue(storageDao.get(user.getLogin()) != null);
+    }
+
+    @Test
+    void register_nullAgeValue_throwsException() {
+        User user = createUser(VALID_LOGIN_VALUE, VALID_PASSWORD_VALUE, null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_invalidAgeValue_throwsException() {
+        User user = createUser(VALID_LOGIN_VALUE, VALID_PASSWORD_VALUE, 10);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_edgeAgeValue_throwsExcepton() {
```

Comment:
There's a typo in the method name: 'register_edgeAgeValue_throwsExcepton' should be 'register_edgeAgeValue_throwsException'.